src/xtgeo/roxutils/_roxar_loader.py

```diff
@@ -0,0 +1,46 @@
+"""Loading rmsapi or roxar module based on availability"""
+
+import importlib
+from typing import Any, Optional
+
+# Roxar*Type are placeholders for the actual types which are not known at this point,
+# and these may be replaced in modules by e.g:
+# ---------
+#     from xtgeo.roxutils._roxar_loader import roxar
+#     if TYPE_CHECKING:
+#         from xtgeo.grid3d.grid_property import GridProperty
+#         if roxar is not None:
+#            from roxar.grids import Grid3D as RoxarGridType
+# ---------
+RoxarType = Any
+RoxarGridType = Any  # rmsapi.grids
+RoxarGrid3DType = Any  # rmsapi.grids.Grid3D
+RoxarWellpicksType = Any  # rmsapi.well_picks
+RoxarJobsType = Any  # rmsapi.jobs
```

Comment:
```suggestion
RoxarType = TypeVar("RoxarType")
RoxarGridType = ForwardRef("rmsapi.grids")
RoxarGrid3DType = ForwardRef("rmsapi.grids.Grid3D")
RoxarWellpicksType = ForwardRef("rmsapi.well_picks")
RoxarJobsType = ForwardRef("rmsapi.jobs")
```
I've not used [ForwardRef](https://docs.python.org/3/library/typing.html#typing.ForwardRef) before, but it might be fit for this purpose. I don't know if this will appease mypy, though

Reply:
Tried something, but not all things are working (or intuitive...)